### ISSUE TITLE
[LargeMarginSoftmax] Could the cos(m * theta) be calculated by torch.cos(m * torch.acos(cosine)) instead of following formula(7) ?

### ISSUE THREAD

**Issue by Mactarvish** [KevinMusgrave/pytorch-metric-learning#352]
I'm digging in the source code of kind of loss functions, and I've noticed that the LargeMarginSoftmaxLoss implemented every formula shown in original paper [https://arxiv.org/pdf/1612.02295.pdf](url) strictly. 
That's exactly no problem, but what I'm wondering is that in the method `get_cos_with_margin` which calculates cos(m * theta) using cos(theta). Why not just calculating theta by `theta = torch.acos(cosine)` and then calculating cos(m*theta) by `torch.cos(self.margin * theta)` but following formula (7) ? The first one seems more Intuitive.

**Comment by KevinMusgrave** [comment]
I can't remember why they used the complicated equation. Did they want to get a specific form when calculating the gradients during backpropagation? I don't think they explain it. It would certainly be a lot cleaner to do ```torch.cos(self.margin * theta)```
